src/System.Diagnostics.PerformanceCounter/src/System/Diagnostics/SharedUtils.cs

```diff
@@ -176,14 +176,7 @@ internal static string GetLatestBuildDllDirectory(string machineName)
 
             try
             {
-                if (machineName.Equals("."))
-                {
-                    return GetLocalBuildDirectory();
-                }
-                else
-                {
-                    baseKey = RegistryKey.OpenRemoteBaseKey(RegistryHive.LocalMachine, machineName);
-                }
+                baseKey = RegistryKey.OpenRemoteBaseKey(RegistryHive.LocalMachine, machineName);
                 if (baseKey == null)
                     throw new InvalidOperationException(SR.Format(SR.RegKeyMissingShort, "HKEY_LOCAL_MACHINE", machineName));
```

Comment:
nit: in the unlikely case that this fails the message: "Cannot open registry key HKEY_LOCAL_MACHINE on computer .." will look a bit odd, albeit is the same one that Fx currently will show...